rootconf/default/bin/discover

```diff
@@ -1,6 +1,7 @@
 #!/bin/sh
```

Comment:
I think this issue needs some more thought.  Also this issue should be in a separate pull request from the lib32/lib64 issues as they are unrelated.

Correct me if I'm wrong, but here is my take on what is happening:

The `sd_dhcp4()` function attempts DHCPv4 on all interfaces, but stops when it gets a reply from one.  Let's say you have two interfaces `eth0` and `eth1`, both of which have a DHCP server (possibility different servers, or different configurations) - lets call the DHCP severs dhcp0 and dhcp1 respectively.

The `sd_dhcp4()` loop will stop after `eth0` gets a DHCP reply from dhcp0.  `eth1` never gets a chance to talk with dhcp1.

In your case the desired DHCP configuration (the tftp server info) is only valid from dhcp1 -- for example the reply from dhcp0 does not contain the tftp server info.

It seems like the ONIE discovery needs to be per interface and the image waterfall driven all the way through before moving on to the next interface.  Today the code is not quite right, where it first tries to gather all the DHCP info from all interfaces and then drive the waterfall.

The reason the code stops after the first interface has a valid reply is that what if both dhcp0 and dhcp1 specify a tftp server?  Who wins?

All this code needs a bit of a rework and you have done most that here -- thanks!

A few comments:

1. `sd_static` and `sd_localfs` should only happen once per discovery loop.  These should not happen once per network interface.  I think once these discoveries are complete we could launch `exec_installer` just for them.

2. `sd_dhcp6`, `sd_dhcp4`, `sd_mdns`, `sd_fallback` and `neigh_discovery` happen once per *interface*.  After all of these network discoveries are complete for an interface, launch `exec_installer`.  This would allow each interface to receive DHCP replies from different servers, with potentially different image URL configuration.

It looks like you have done most of this already.  The only change I would make is that the static and local file system discovery only happen once per "discovery loop", instead of once per interface.  Otherwise looks pretty reasonable.